src/urllib3/util/connection.py

```diff
@@ -1,11 +1,15 @@
 import socket
+from typing import List, Optional, Tuple
 
 from urllib3.exceptions import LocationParseError
 
 from .wait import wait_for_read
 
+SOCKET_GLOBAL_DEFAULT_TIMEOUT = socket._GLOBAL_DEFAULT_TIMEOUT  # type: ignore
+SocketOptions = List[Tuple[int, int, int]]
```

Comment:
Should this be restricted to tuples of ints or the last value could also be str/bytes?

Reply:
The type for `socket.socket.setsockopt()` appears to be `Revealed type is 'def (self: socket.socket, level: builtins.int, optname: builtins.int, value: Union[builtins.int, builtins.bytes])` so do `int/bytes` for last type.

Reply:
Changed to `List[Tuple[int, int, Union[int, bytes]]]`